Change appearance of speed labels in status bar

buinsky opened the pull request:
Status bar befor:
![before](https://cloud.githubusercontent.com/assets/9297293/12015359/4567581a-ad4c-11e5-8d03-cf7e13d51a61.png)

Status bar after:
![after](https://cloud.githubusercontent.com/assets/9297293/12015361/4a947ee4-ad4c-11e5-88da-865a17a69443.png)

buinsky commented:
@glassez Please check the code.

sledgehammer999 commented:
@buinsky thanks for this code and effort, but personally I don't like this UI change.
Also, if I remember correctly, this was done in order to follow the same scheme used by utorrent. I can't remember you implemented it, I think it was @Chocobo1.

Unless, there is strong opposition from other people I am going to reject this UI change. It takes more space and IMO is more confusing.
@qbittorrent/qbittorrent-frequent-contributors

buinsky commented:
@qbittorrent/qbittorrent-infrequent-contributors What you think?

IMHO current variant is more confusing, It's hard to determine what the current speed is at first sight.

glassez commented:
I don't like these changes. The current version is more concise. The only thing I would change in this is the order of the values. I think it should be: Speed, Speed Limit, Total Up-/Downloaded.

buinsky commented:
@sledgehammer999 What if only speed limits will be separated. Do you like such option?

buinsky commented:
@sledgehammer999 It will be look like this:

![after](https://cloud.githubusercontent.com/assets/9297293/12028390/e9bc22e2-ade7-11e5-873e-d264804cc0f4.png)

naikel commented:
I actually like the current UI.  Maybe I would change the order like @glassez said but what I'd definitely do is to show them in different colors (like red for the limits and things like that) to read them more easily.  Something like:

```100 KiB/s [Limit: 500 KiB/sec] (20.0 GiB)```

With [Limit: 500 KiB/sec] all in red and (20.0 GiB) all in blue :grin:

Chocobo1 commented:
>Also, if I remember correctly, this was done in order to follow the same scheme used by utorrent. I can't remember who implemented it, I think it was @Chocobo1.

 :trollface: not me.

---

>100 KiB/s \[Limit: 500 KiB/sec\] (20.0 GiB)
>With [Limit: 500 KiB/sec] all in red and (20.0 GiB) all in blue 

:+1:  for this, easier to read.
@buinsky , if you're willing to impl it. Use the blue color in [speed widget](https://github.com/qbittorrent/qBittorrent/issues/4134) and find a suitable red.

I wonder how these color variations are chosen?

sledgehammer999 commented:
Well, from git history it implemented by @pmzqla (check #2061 and #1316).
Using colors is very a good idea generally. But I don't know what will happen on linux where users can have dark themes enabled...

Ideas?

buinsky commented:
@sledgehammer999 We can analyse background color. If it is bright we use colors set 1. If it is dark we use colors set 2.

sledgehammer999 commented:
>We can analyse background color. If it is bright we use colors set 1. If it is dark we use colors set 2.

Oh yes, like I do in the transferlist.
Maybe use the same "salmon" for red as for `TorrentState::PausedUploading` and the different blues for `TorrentState::ForcedUploading` ?

buinsky commented:
@sledgehammer999 I changed the code. Now the labels looks like this:
![result](https://cloud.githubusercontent.com/assets/9297293/12171245/3e2cb3f0-b559-11e5-9b41-970516043f94.png)
on dark background:
![result2](https://cloud.githubusercontent.com/assets/9297293/12171249/4370798c-b559-11e5-9115-2947aa991b7e.png)

Do you like the result?

buinsky commented:
May be the red color is distracting?

glassez commented:
@buinsky, if @sledgehammer999 will take this changes, don't forget to fix the coding style.

ngosang commented:
I'm opposed. :-1:

buinsky commented:
> I'm opposed. :-1:

@ngosang Why?

ngosang commented:
Too many colors in the status bar, is distracting me.

buinsky commented:
I can remove colors from labels. @sledgehammer999 What will you say?

naikel commented:
:+1: for colors

We already have colors, for example active seeding torrents are blue. IMHO they are not distracting at all.

LordNyriox commented:
@buinsky @sledgehammer999 Personally, I like the recolor.  It helps to visually separate the limits from the current values.

naikel commented:
Maybe an advanced option to turn off the status bar colors for people that
don't like it? or is it too much?

On Tue, Jan 12, 2016 at 12:49 PM, LordNyriox <notifications@github.com>
wrote:

> @buinsky <https://github.com/buinsky> @sledgehammer999
> <https://github.com/sledgehammer999> Personally, I like the recolor. It
> helps to visually separate the limits from the current values.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qbittorrent/qBittorrent/pull/4463#issuecomment-170980559>
> .
>

buinsky commented:
@sledgehammer999 What you think about the advanced option?

DrKittens commented:
How is this less complicated to read?
It just looks extremely messy, can we have an option in advanced/normal settings to use legacy display?
Im quite attached to the "Download: [Limit]CurrentSpeed | Upload" [Limit]CurrentSpeed" Layout

http://i.imgur.com/XCTUxdP.png

buinsky commented:
I removed coloring from the PR. The changes I made:

1) I changed the order of the values now it: Speed, Speed Limit, Total Up-/Downloaded (as adviced @glassez)
2) Now content of the labels are left aligned (like in uTorrent) because when it center aligned the speed may twitch.
3) I modify friendlyUnit function to cut fractional part of numbers from 1000 to 1023. It helps to save space and helps to save columns width in WebUI.

I don't what to release coloring of the speed labels. If somebody wants to, he can view my commit # [de95fa523968e98568079d23df8d576f029c920b](https://github.com/buinsky/qBittorrent/commit/de95fa523968e98568079d23df8d576f029c920b) .

@sledgehammer999 You are free to merge the PR.

sledgehammer999 commented:
Sorry for the long delay. Coloring **sounded** good but in practice it looks out of place.
I can live with the new re-ordered list of values. I never look at them anyway.

About the friendly unit function. Can you explain more why you modified it? I don't understand the need. Maybe an example?

buinsky commented:
> About the friendly unit function. Can you explain more why you modified it? I don't understand the need. Maybe an example?

@sledgehammer999  Speeds from 1000.0 to 1023.9 appears very rarely. But when the speed jumps about this interval +/- all the columns in the torrents table in WebUI periodically change their width. It's annoying. I'm not very good at html to fix this another way. Also shorter numbers are easier to read. I can leave that only for WebUI as workaround.

![default](https://cloud.githubusercontent.com/assets/9297293/12325770/a22f9622-badd-11e5-9d7a-a533db9a96dd.png)

glassez commented on src/gui/statusbar.cpp:
```diff
@@ -42,225 +42,256 @@
+    m_upSpeedLbl->setFlat(true);
+    m_upSpeedLbl->setFocusPolicy(Qt::NoFocus);
+    m_upSpeedLbl->setCursor(Qt::PointingHandCursor);
+    m_upSpeedLbl->setStyleSheet("text-align:left;");
```
Is it necessary?

buinsky replied:
Yes.

glassez replied:
Why?
1. How the label is aligned by default? 
2. Isn't there a corresponding function to set text alignment?

buinsky replied:
1. By default QLabel is left-aligned. But it is not QLabel. ```m_upSpeedLbl``` is QPushButton stylized like label.
2. No.

glassez replied:
OK 👌

glassez commented on src/gui/statusbar.cpp:
```diff
@@ -42,225 +42,256 @@
+    QLabel *restartIconLbl = new QLabel(m_bar);
+    restartIconLbl->setPixmap(QPixmap(":/icons/oxygen/dialog-warning.png").scaled(QSize(24,24)));
+    restartIconLbl->setToolTip(restartText);
+    m_bar->insertWidget(0,restartIconLbl);
```
Space after comma.

buinsky replied:
Fixed.

buinsky commented:
@sledgehammer999 Please merge the PR.

ngosang commented:
:+1:

LordNyriox commented:
@buinsky: Please rebase this PR at some point. 

Thanks, 
~ Nyriox

buinsky commented:
Rebased.

sledgehammer999 commented:
Your code is ok(apart from my comment). However, I feel there is a lot of space wasted in the speed labels. I think they look weird.

Take a look:
![untitled](https://cloud.githubusercontent.com/assets/273315/13751300/32552060-ea12-11e5-852e-860961edb794.jpg)

What do you think?

sledgehammer999 commented:
This can be fixed if by deleting line 131 in  statubar.cpp (`m_container->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);`)
But in this case the speed labels need a little bit of padding.
What do you think?

buinsky commented:
@sledgehammer999 

I removed the line 131 and set minimum width of the speed labels to 200. Now the status bar looks like this:

![speed_labels](https://cloud.githubusercontent.com/assets/9297293/13787737/39b2bad4-eaee-11e5-88e6-ef75badbd5fd.png)

sledgehammer999 commented:
This seems good. Thank you.
